laurat007/assignment1/Q1.java

```diff
@@ -0,0 +1,120 @@
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.StringTokenizer;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author laura
+ */
+public class Q1 {
```

Comment:
Okay, you basically refactored all the previous content of the main method into a new large method. That is better but that is not exactly what I had in mind :)

Can you structure your code so that the solve method looks like that:
1. read input file
2. iterate through N test cases and depending on the value of identifier evoke one of three methods:
   a. caseSensitiveAnagrams(String s1, String s2)
   b. caseInsensitiveAnagrams(String s1, String s2)
   c. caseInsensitiveSentenceAnagrams(String s1, String s2)

The benefit in structuring your code like that is that each one of those functions could potentially be called when the two strings are provided from another source than an input file. For example you could then easily write unit test suite and test each method.

Reply:
Done three different methods and restructured the code.